File: frontend/client/src/types/activity.ts

import z from "zod";

export const activityObject = z.object({
  "id": z.string(),
  "title": z.string().min(2).max(200),
  "description": z.string().min(2).max(500),
  "eventDate": z.string(),
  "category": z.string().nonempty(),
  "isCancelled": z.coerce.boolean(),
  "city": z.string().min(2).max(500),
  "venue": z.string().min(2).max(500),
  "latitude": z.coerce.number({
    error: "Latitude should be a number"
  }).min(-90).max(90),
  "longitude": z.coerce.number({
    error: "Longitude should be a number"
  }).min(-180).max(180)
});

export const activityResponseObject = z.object({
  "errorMessage": z.string().nullable(),
  "value": z.array(activityObject),
  "status": z.boolean(),
  "errorCode": z.number().nullable()

});

export type ActivityResponse = z.infer<typeof activityResponseObject>;
export type Activity = z.infer<typeof activityObject>;

File: backend/Domain/Models/Activity.cs

namespace Domain.Models
{
    public class Activity : BaseModel
    {
        public required string Title { get; set; }
        public required string Description { get; set; }
        public required DateTime EventDate { get; set; }
        public required string Category { get; set; }
        public required bool IsCancelled { get; set; }

        //location properties
        public required string City { get; set; }
        public required string Venue { get; set; }
        public double Latitude { get; set; }
        public double Longitude { get; set; }

        public virtual ICollection<Attendees> Attendees { get; set; } = [];

        public virtual ICollection<Comment> Comments { get; set; } = [];
    }
}